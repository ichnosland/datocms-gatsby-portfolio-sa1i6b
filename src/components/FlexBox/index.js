/**
*
* FlexBox
*
*/
import styled, { css } from 'styled-components';

import Div from '../Div';
import media from 'style/mediainjector';

export const FlexBox = styled(Div)`
  display: flex;
  flex-direction: ${(props) => props.direction ? props.direction : 'row'};
  justify-content: ${(props) => props.justifyContent ? props.justifyContent : 'center'};
  align-items: ${(props) => props.alignItems ? props.alignItems : 'center'};
  align-content: ${(props) => props.alignContent ? props.alignContent : 'center'};
  flex-wrap: ${(props) => props.wrap ? props.wrap : 'nowrap'};
  ${media.print`
   ${({ noPrint }) => noPrint && css`
       display:none;
   `}
 `}
`;

export const FlexChild = styled(FlexBox)`
  order: ${(props) => props.order};
  flex-grow: ${(props) => props.flexGrow};
  flex-basis: ${(props) => props.flexBasis};
  align-self: ${(props) => props.alignSelf};
`;

export const FlexWrap = styled(FlexBox)`
  position: relative;
  flex-direction: column;
  justify-content: space-between;
  min-height: 100%;
  height: 100%;
  ${media.desktop`
    &:before {
      content: '';
      display: block;
    }
  `}
  ${media.mobile`
    display: block;
  `}
`;

export default FlexBox;
