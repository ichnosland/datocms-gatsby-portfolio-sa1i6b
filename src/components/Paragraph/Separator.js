/**
*
* Separator
*
*/


import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import FlexWrap from 'components/FlexBox';

export const SeparatorBg = styled.div`
  background-image: url('data:image/svg+xml;utf8,<svg width="4" height="4" viewBox="0 0 4 4" xmlns="http://www.w3.org/2000/svg"><path fill="${(props) => props.fill ? props.fill : props.theme.light}" d="M0 0h4v4H0z" fill-rule="evenodd"/></svg>');
  background-size: 1px 1px;
  background-position: center;
  background-repeat: repeat-x;
  flex-grow: 1;
  align-self: stretch;
`;

export const SeparatorText = styled.span`
  color: ${(props) => props.color ? props.color : props.theme.light};
  display: inline-block;
  padding: 0.5em;
  position: relative;
`;

class Separator extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    const { text, color } = this.props;
    return (
      <FlexWrap>
        <SeparatorBg fill={color} />
        <SeparatorText color={color}>{text}</SeparatorText>
        <SeparatorBg fill={color} />
      </FlexWrap>
    );
  }
}

Separator.propTypes = {
  color: PropTypes.string,
  text: PropTypes.string,
};

export default Separator;
