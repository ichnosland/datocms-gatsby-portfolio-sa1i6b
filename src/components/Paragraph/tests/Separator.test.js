import React from 'react';
import { shallow } from 'enzyme';

import FlexBox from 'components/FlexBox';
import Separator, { SeparatorBg, SeparatorText } from '../Separator';

const theme = {
  brand: '#00BBEF',
  light: '#1588AA',
};

describe('<Separator />', () => {
  it('should render a FlexBox tag', () => {
    const renderedComponent = shallow(<Separator />);
    expect(renderedComponent.find(FlexBox).length).toEqual(1);
  });

  it('should render its css w/ props color', () => {
    const color = 'red';
    const renderedComponent = shallow(<Separator color={color} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props text', () => {
    const text = 'Testo';
    const renderedComponent = shallow(<Separator text={text} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props theme', () => {
    const renderedComponent = shallow(<Separator theme={theme} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});

describe('<SeparatorText />', () => {
  it('should render a <span> tag', () => {
    const renderedComponent = shallow(<SeparatorText />);
    expect(renderedComponent.type()).toEqual('span');
  });

  it('should render its css w/ props color', () => {
    const color = 'yellow';
    const renderedComponent = shallow(<SeparatorText color={color} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props theme', () => {
    const renderedComponent = shallow(<SeparatorText theme={theme} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});

describe('<SeparatorBg />', () => {
  it('should render a <div> tag', () => {
    const renderedComponent = shallow(<SeparatorBg />);
    expect(renderedComponent.type()).toEqual('div');
  });

  it('should render its css w/ props fill', () => {
    const color = 'green';
    const renderedComponent = shallow(<SeparatorBg fill={color} />);
    const SepBg = renderedComponent.find('div');
    const SepBgFill = SepBg.prop('fill');
    expect(SepBgFill).toBe('green');
  });
});
