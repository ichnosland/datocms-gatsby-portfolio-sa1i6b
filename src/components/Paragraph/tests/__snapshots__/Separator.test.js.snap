// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Separator /> should render its css w/ props color 1`] = `
<FlexBox>
  <Separator__SeparatorBg
    fill="red"
  />
  <Separator__SeparatorText
    color="red"
  />
  <Separator__SeparatorBg
    fill="red"
  />
</FlexBox>
`;

exports[`<Separator /> should render its css w/ props text 1`] = `
<FlexBox>
  <Separator__SeparatorBg />
  <Separator__SeparatorText>
    Testo
  </Separator__SeparatorText>
  <Separator__SeparatorBg />
</FlexBox>
`;

exports[`<Separator /> should render its css w/ props theme 1`] = `
<FlexBox>
  <Separator__SeparatorBg />
  <Separator__SeparatorText />
  <Separator__SeparatorBg />
</FlexBox>
`;

exports[`<SeparatorText /> should render its css w/ props color 1`] = `
.c0 {
  color: yellow;
  display: inline-block;
  padding: 0.5em;
  position: relative;
}

<span
  className="c0"
  color="yellow"
/>
`;

exports[`<SeparatorText /> should render its css w/ props theme 1`] = `
.c0 {
  color: #1588AA;
  display: inline-block;
  padding: 0.5em;
  position: relative;
}

<span
  className="c0"
/>
`;
